syndeo-template/README.md

```diff
@@ -18,46 +18,95 @@ transforms. Then, the template will take the pipeline spec, and
-  -Dexec.args="--runner=DataflowRunner --project=$GCP_PROJECT --region=$REGION \
-               --gcpTempLocation=$GCS_BUCKET --pipelineSpec=$PATH_TO_SPEC --tempLocation=$GCS_BUCKET/temp"
+gcloud config set project PROJECT
+```
+
+#### BigQuery to BigTable tests
+
+The file `BigTableWriteIT` **holds integration tests** for the basic BigTable Syndeo integration. These integration tests
+rely on the existence of a BigTable instance and table, as well as a BigQuery dataset, which holds the BQ data that
+is part of the pipeline's read.
+
+This integration test requires a BigQuery dataset and a BigTable instance, as well as GCS buckets to handle artifacts.
+
+1. Create the BigQuery dataset.
+
+```
+bq mk syndeo_dataset
+```
+
+2. Create the BigTable instance.
+
+```
+gcloud bigtable instances create teleport --display-name=teleport --cluster-config=id=teleport,zone=us-central1-a
+```
+
+3. Create the artifact bucket
 
-# Running a pushed template (see Push template below)
-Run Template on Dataflow from Uploaded Template:
-gcloud dataflow flex-template run "syndeojob-`date +%Y%m%d-%H%M%S`" \
-  --template-file-gcs-location "gs://$GCS_BUCKET/template.json" --region $REGION \
-  --temp-location "gs://$GCS_BUCKET/temp" --parameters=pipelineSpec="gs://$GCS_BUCKET/pubsub_to_avro_config.txt"
 ```
+ARTIFACT_BUCKET=[CHANGE ME]
+gsutil mb gs://$ARTIFACT_BUCKET
+```
+
+4. Create the temporary location bucket
+
+```
+TEMP_LOCATION_BUCKET=[CHANGE ME]
+gsutil mb gs://$TEMP_LOCATION_BUCKET
+```
+
+5. Run the end-to-end test.
+
+The below command tests the workflow by publishing some data to BQ and checking that it makes it over to Bigtable.
+
+
+```shell
+mvn clean package test -f pom.xml -pl syndeo-template/pom.xml  \
+    -Dtest="BigTableWriteIT#testBigQueryToBigTableSmallNonTemplateJob"  \
+    -Dproject="$(gcloud config get-value project)"  -DartifactBucket="gs://$ARTIFACT_BUCKET"  \
+    -Dregion="us-central1" -DtempLocation=gs://$TEMP_LOCATION_BUCKET
+```
+
+### Push template to GCP
+
+The following command will take the already-built artifacts and push them to GCS and GCR, where they can be utilized to run a template.
+
+(Note: The project does not currently have an end-to-end test to validate template-based runs because the template would run as part of
+syndeo, which involves repeated runs and so on).
 
-Push template to GCP
 ```shell
 # Push Template to GCP:
-gcloud dataflow flex-template build gs://$GCS_BUCKET/template.json \
-    --sdk-language "JAVA"  --flex-template-base-image JAVA11 \
-    --image-gcr-path='gcr.io/$GCP_PROJECT/template:latest' \
-    --jar "target/syndeo-template-1.0-SNAPSHOT.jar" \
+gcloud dataflow flex-template build gs://$GCS_BUCKET_NAME/syndeo-template.json  \
+    --metadata-file syndeo-template/metadata.json   --sdk-language "JAVA"  \
+    --flex-template-base-image JAVA11     --image-gcr-path=gcr.io/$GCP_PROJECT/syndeo-template:latest \
+    --jar "syndeo-template/target/syndeo-template-1.0-SNAPSHOT.jar"  \
```

Comment:
Running this command yielded an error about `No such file or directory: syndeo-template/target/syndeo-template-1.0-SNAPSHOT.jar` which I tried to solve by running `mvn package` but this led to `bigquery-to-bigtable: Missing target image parameter, perhaps you should add a <to><image> configuration parameter to your pom.xml or set the parameter via the commandline (e.g. 'mvn compile jib:build -Dimage=<your image name>')`.

Reply:
I reproduced this issue. Thanks Damon! I added the proper package command.